app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx

```diff
@@ -190,3 +206,46 @@ const setDeepLinkHandler = (
  * email, then the URL will be for the confirmation page path. */
 const isConfirmRegistrationRedirect = (url: URL) =>
   url.pathname === CONFIRM_REGISTRATION_PATHNAME;
+
+/** If the user is being redirected after a sign-out, then query args will be present. */
+const isSignInRedirect = (url: URL) =>
+  url.pathname === SIGN_IN_PATHNAME && url.search !== "";
+
+/** When the user is being redirected from a federated identity provider, then we need to pass the
+ * URL to the Amplify library, which will parse the URL and complete the OAuth flow. */
+const handleAuthResponse = (url: string) => {
+  void (async () => {
+    /** Temporarily override the `window.location` object so that Amplify doesn't try to call
+     * `window.location.replaceState` (which doesn't work in the renderer process because of
+     * Electron's `webSecurity`). This is a hack, but it's the only way to get Amplify to work
+     * with a custom URL protocol in Electron.
+     *
+     * Note that this entire hack must happen within an async IIFE block, because we need to be
+     * sure to restore the original `window.location.replaceState` function before any
+     * non-Amplify code runs, which we can't guarantee if `_handleAuthResponse` is allowed to
```

Comment:
`_handleAuthResponse` still completes asynchronously - `await` queues the rest of the function in the event loop, so other functions will run while `_handleAuthResponse` is waiting

Reply:
Good catch. This doc is mistaken. That being said, I think it was unnecessary in the first place. Removing and resetting the function works, so perhaps the `await` is fine.